cmd/crank/beta/lint/cmd.go

```diff
@@ -0,0 +1,203 @@
+*/
+
+// Package lint implements lint command for Composite Resource Definitions (XRDs).
+package lint
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/alecthomas/kong"
+	"gopkg.in/yaml.v3"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	sigyaml "sigs.k8s.io/yaml"
+
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+
+	v1 "github.com/crossplane/crossplane/apis/apiextensions/v1"
+)
+
+// Cmd arguments and flags for render subcommand.
+type Cmd struct {
+	// Arguments.
+	Resources string `arg:"" help:"Resources source which can be a file, directory, or '-' for standard input."`
+
+	// Flags.
+	Output             string `default:"stdout"                      help:"Output format. Valid values are 'stdout' or 'json'. Default is 'stdout'." short:"o"`
+	SkipSuccessResults bool   `help:"Skip printing success results."`
+}
+
+// Help prints out the help for the validate command.
+func (c *Cmd) Help() string {
+	return `
+This command checks the provided Crossplane CompositeResourceDefinition (XRD) via stdin, files or directories of such files for 
+API Design Best Practices. It scans each resource and identifies issues such as missing descriptions, required fields and more.
+
+The command will return a non-zero exit code if any errors are found. The exit codes are as follows:
+  - 0: No issues found
+  - 1: Errors found
+  - 2: Warnings found
+
+Examples:
+
+  # Lint a single XRD file 
+  crossplane beta lint xrd.yaml
+
+  # Lint all XRD files in the directory "xrds"
+  crossplane beta lint ./xrds/
+
+  # Lint all XRD files in the directory "xrds" and skip success logs
+  crossplane beta validate ./xrds/ --skip-success-results
+
+  # Lint all XRD files in the directory "xrds" and output results in JSON format
+  crossplane beta lint ./xrds/ --output json
+`
+}
+
+// Issue represents a linting issue found in a resource.
+type Issue struct {
+	RuleID  string `json:"id"`
+	Name    string `json:"name"`
+	Line    int    `json:"line"`
+	Error   bool   `json:"error"`
+	Message string `json:"message"`
+}
+
+type output struct {
+	Summary struct {
+		Valid    bool `json:"valid"`
+		Total    int  `json:"total"`
+		Errors   int  `json:"errors"`
+		Warnings int  `json:"warnings"`
+	} `json:"summary"`
+	Issues *[]Issue `json:"issues"`
+}
+
+// Run lint.
+func (c *Cmd) Run(k *kong.Context, _ logging.Logger) error {
+	// Load all resources
+	resourceLoader, err := NewLoader(c.Resources)
+	if err != nil {
+		return errors.Wrapf(err, "cannot load resources from %q", c.Resources)
+	}
+
+	resources, err := resourceLoader.Load()
+	if err != nil {
+		return errors.Wrapf(err, "cannot load resources from %q", c.Resources)
+	}
+
+	var allIssues []Issue
+
+	for _, resource := range resources {
+		resourceAsBytes, err := yaml.Marshal(resource)
+		if err != nil {
+			return errors.Wrapf(err, "cannot convert to yaml %q", resource.Value)
+		}
+
+		var k8s unstructured.Unstructured
```

Comment:
```suggestion
		var xrd unstructured.Unstructured
```

I'd prefer this name - better describes what the unstructured resource is.